### PR TITLE
update sim_coinselect.py to allow different output types

### PR DESCRIPTION
**Pull request by josibake**
used for specifying weights for each address type. this is for testing #24584

### CODE REVIEW

File: test/functional/sim_coinselect.py [hunk 1]
```diff
@@ -125,6 +145,7 @@ def add_options(self, parser):
         parser.add_argument("--payments", default=None, required="--feerates" in sys.argv, action=PaymentsFeeratesOptionsAction)
         parser.add_argument("--feerates", default=None, required="--payments" in sys.argv, action=PaymentsFeeratesOptionsAction)
         parser.add_argument("--ops", type=int, default=None)
+        parser.add_argument("--weights", type=int, nargs='+', default=None, required="--weights" in sys.argv, action=OutputTypeWeightsOptionsAction, help="Weight values for each address type. Weights must add to 100 and be provided in the following order: bech32m bech32 p2sh-segwit legacy")
```
Comment (Xekyo) [inline test/functional/sim_coinselect.py:148]: Let's be a bit clearer what this does in the first place:

```suggestion
        parser.add_argument("--weights", type=int, nargs='+', default=None, required="--weights" in sys.argv, action=OutputTypeWeightsOptionsAction, help="Causes recipient output types to be chosen randomly with provided weights per address type. Weights must add to 100 and be provided in the following order: bech32m bech32 p2sh-segwit legacy")
```

**Review by Xekyo — COMMENTED** [review]
I'm not sure I grok all of the things you did with the inp fee. I'll have to look at that later again, but for know just the note that your help text doesn't actually say what the option does. :)

**Comment by josibake** [comment]
> I'm not sure I grok all of the things you did with the inp fee

it's a move-only change (or should be :sweat_smile: ). previously, the code relied on there only being witness txs. since the simulation now needs to support non_witness_tx, i rearranged so we get the amount once, from witness or non_witness input, and then we do the inp fee calculation

File: test/functional/sim_coinselect.py (hunk 1 above)
Comment (josibake) [reply test/functional/sim_coinselect.py]: added your wording, thx!

**Comment by achow101** [comment]
picked and squashed into the branch.

**Pull request closed without merging**